### PR TITLE
Chain extension

### PR DESCRIPTION
**Pull request by fbielejec**
Adds a basic chain extension, allowing to make a call to the randomness pallet from a contract. Could be extended to any arbitrary calls.

*NOTE*
Technically the same could be achieved with `call_runtime` but the purpose is to have a starting point for more interesting extensions (e.g. `call_runtime` cannot be used for storage queries).

### CODE REVIEW

**Comment by JoshOrndorff** [comment]
Looks interesting. Please add a description with some context when you are ready.

File: runtime/src/chain_extension.rs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        }
+    }
+}
+
+impl AcademyPowChainExtension {
+    fn random<E: Ext>(env: Environment<E, InitState>) -> Result<RetVal, DispatchError> {
+        debug!(
+            target: "runtime",
+            "[ChainExtension]|call|func_id:{:}",
+            RANDOM_FUNCTION_OK
```
Comment (pmikolajczyk41) [inline runtime/src/chain_extension.rs:43]: RANDOM_FUNCTION_ID

File: runtime/src/chain_extension.rs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        );
+
+        let mut env = env.buf_in_buf_out();
+
+        let arg: [u8; 32] = env.read_as()?;
+        let random_seed = RandomnessCollectiveFlip::random(&arg).0;
```
Comment (pmikolajczyk41) [inline runtime/src/chain_extension.rs:49]: you are missing charging appropriate amount of gas - unfortunately for chain extensions we have to do it manually; i guess that taking corresponding weight from pallet rcf should be fine

**Review by pmikolajczyk41 — CHANGES_REQUESTED** [review]

**Review by JoshOrndorff — APPROVED** [review]
I'm fine with this whenever @pmikolajczyk41 and @fbielejec think it is good enough. I'm not qualified to review it technically.

**Comment by JoshOrndorff** [comment]
Closing this because we will not run contracts on the pow network this time.

**Pull request closed without merging**